ResearchKit/ActiveTasks/ORKRecorder.m

```diff
@@ -274,7 +274,7 @@ @implementation ORKDataLogRecorder {
 @synthesize logger = _logger;
 
 - (ORKDataLogger *)logger {
-    return self.sharedLogger ? : _logger;
+    return [self isConsolidated] && (self.sharedLogger != nil) ? self.sharedLogger : _logger;
```

Comment:
isn't the (self.sharedLogger != nil) redundant? or do you still want to check that just in case the isConsolidated method is overridden and no longer implies that?